lib/expr.ml

```diff
@@ -345,12 +345,10 @@ let triop' (ty : Ty.t) (op : triop) (e1 : t) (e2 : t) (e3 : t) : t =
 [@@inline]
 
 let triop ty (op : triop) (e1 : t) (e2 : t) (e3 : t) : t =
-  match (view e1, view e2, view e3) with
-  | Val v1, Val v2, Val v3 -> value (Eval.triop ty op v1 v2 v3)
-  | Val v, _, _ -> (
-    match op with
-    | Ite -> ( match v with True -> e2 | False -> e3 | _ -> assert false )
-    | _ -> triop' ty op e1 e2 e3 )
+  match (op, view e1, view e2, view e3) with
+  | op, Val v1, Val v2, Val v3 -> value (Eval.triop ty op v1 v2 v3)
+  | Ite, Val True, _, _ -> e2
+  | Ite, Val False, _, _ -> e3
```

Comment:
Should'nt they be in this order:

```ocaml
  | Ite, Val True, _, _ -> e2
  | Ite, Val False, _, _ -> e3
  | op, Val v1, Val v2, Val v3 -> value (Eval.triop ty op v1 v2 v3)
```

Also you can merge the two cases like this:

```ocaml
  | (Ite, Val True, e, _) | (Ite, Val False, _, e) -> e
  | op, Val v1, Val v2, Val v3 -> value (Eval.triop ty op v1 v2 v3)
```

Reply:
> Should'nt they be in this order:
> 
> ```ocaml
>   | Ite, Val True, _, _ -> e2
>   | Ite, Val False, _, _ -> e3
>   | op, Val v1, Val v2, Val v3 -> value (Eval.triop ty op v1 v2 v3)
> ```
> Also you can merge the two cases like this:
> 
> ```ocaml
>   | Ite, Val True, e, _ | Ite, Val False, _, e -> e
>   | op, Val v1, Val v2, Val v3 -> value (Eval.triop ty op v1 v2 v3)
> ```

Indeed, I will fix it :smile: Thanks again!